kythe/typescript/indexer.ts

```diff
@@ -1549,12 +1565,31 @@ class Visitor {
    *     ((local var)|const|let) <name>: <type>( = <initializer>)?
    * where `(local var)` is the declaration of a variable in a catch clause.
    */
-  emitVariableDeclarationCode(decl: ts.VariableDeclaration, declVName: VName) {
+  emitVariableDeclarationCode(decl: ts.VariableDeclaration|ts.BindingElement, declVName: VName) {
     const codeParts: MarkedSource[] = [];
-    const initializerList = decl.parent;
+    let varDecl;
+    const bindingPath: Array<string|number> = [];
+    if (ts.isBindingElement(decl)) {
+      // The node we want to emit code for is a BindingElement. This parent of
+      // this is always a BindingPattern; the parent of the BindingPattern is
+      // another BindingPattern, a ParameterDeclaration, or VariableDeclaration.
+      // We handle ParameterDeclarations in `visitParameters`, so here we only
+      // care about the declaration from a VariableDeclaration.
+      bindingPath.push(bindingElemIndex(decl));
+      varDecl = decl.parent.parent;
+      while (!ts.isVariableDeclaration(varDecl)) {
+        if (ts.isParameter(varDecl)) return;
+        bindingPath.push(bindingElemIndex(varDecl));
+        varDecl = varDecl.parent.parent;
```

Comment:
Is it possible there's some unpredictable layout here that will cause this loop to never terminate, or to run into an undefined?

Reply:
No, the types save us from that. The type of `BindingElement#parent#parent` is `VariableDeclaration|BindingElement|ParameterDeclaration`. So eventually we will run into a variable or param decl, both of which are handled here. I'll add an undefined check just in case though.